test/e2e/proxy_test.go

```diff
@@ -32,6 +34,44 @@ func proxyTestSuite(t *testing.T) {
 
 	t.Run("Trusted CA ConfigMap validation", testTrustedCAConfigMap)
 	t.Run("Environment variables validation", testEnvVars)
+	t.Run("Certificate validation", testCerts)
+}
+
+func testCerts(t *testing.T) {
+	tc, err := NewTestContext()
+	require.NoError(t, err)
+
+	cm, err := tc.client.K8s.CoreV1().ConfigMaps(wmcoNamespace).Get(context.TODO(),
+		certificates.ProxyCertsConfigMap, meta.GetOptions{})
+	require.NoErrorf(t, err, "error getting trusted CA ConfigMap: %w", err)
+
+	// Read all certs from CM data
+	trustedCABundle := cm.Data[certificates.CABundleKey]
+	assert.Greater(t, len(trustedCABundle), 0, "no additional user-provided certs in bundle")
+
+	certs := x509.NewCertPool()
+	require.True(t, certs.AppendCertsFromPEM([]byte(trustedCABundle)), "unable to parse certs from trusted CA ConfigMap data")
+	subjects := certs.Subjects()
+	// Ensure each cert has been imported into every Windows instance's system store
+	for _, node := range gc.allNodes() {
```

Comment:
Where is `gc.allNodes` being set?
What happens if a developer just wants to run this individual test?

Reply:
I see, I'll call `tc.loadExistingNodes()` at the start of the proxy test suite so we can use gc.allNodes() safely

We might need to do the same in the storage and network test suites, could you confirm my understanding? If so that can be done in a followup PR

Reply:
yeah either tc.loadExistingNodes() or waitForWindowsNodes needs to be called. Things could be a bit more consistent, but both achieve the same thing of populating the nodes within gc.